app/models/manageiq/providers/openstack/infra_manager/host.rb

```diff
@@ -439,4 +441,10 @@ def refresh_network_interfaces(ssu)
   def self.display_name(number = 1)
     n_('Host (OpenStack)', 'Hosts (OpenStack)', number)
   end
+
+  def update_create_event
+    events = ext_management_system.ems_events.where(:event_type => "compute.instance.create.end")
```

Comment:
I'd suggest filter event_streams records with not null host_id, otherwise looks good to me and hopefully should be fast since it covers only undercloud Host and fields we're querying (ems_id, event_type and host_id) are indexed.